src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -774,6 +779,56 @@ export const selectPortfolioBridgeAssets = createDeepEqualOutputSelector(
   },
 )
 
+export const selectPortfolioAnonymized = createDeepEqualOutputSelector(
+  selectAssets,
+  selectWalletId,
+  selectWalletName,
+  selectPortfolioFiatBalances,
+  (assetsById, walletId, walletName = '', portfolioBalances): AnonymizedPortfolio => {
+    const hashedWalletId = hashCode(walletId || '')
+
+    type AssetBalances = Record<string, string>
+    type ChainBalances = Record<string, string>
+
+    const [assetBalances, chainBalances, portfolioBalanceBN] = Object.entries(
+      portfolioBalances,
+    ).reduce<[AssetBalances, ChainBalances, BigNumber]>(
+      (acc, [assetId, balance]) => {
+        // by asset
+        const assetName = getMaybeCompositeAssetSymbol(assetId, assetsById)
+        acc[0][assetName] = balance
+
+        // by chain
+        const { chainId } = fromAssetId(assetId)
+        const chain = getChainAdapterManager().get(chainId)?.getDisplayName()
+        if (!chain) return acc
```

Comment:
this return seems undesirable? would still want to try to do total balance calc even if we don't have chain adapter name for some reason?

Reply:
yeah, not realistic scenario. could still `if (chain) do stuff` instead of return just to be a bit more "correct" imo, but again not huge deal either way